### ISSUE TITLE
Hilt -> error: cannot access DefaultActivityViewModelFactory after updating to 2.34 and 23.4.1

### ISSUE THREAD

**Issue by catalin-florescu** [google/dagger#2548]
After update when I build the project i got this error

    error: cannot access DefaultActivityViewModelFactory
    class file for dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory not found
    Consult the following stack trace for details.
    cannot access DefaultActivityViewModelFactory
    com.sun.tools.javac.code.Symbol$CompletionFailure: class file for dagger.hilt.android.internal.lifecycle.DefaultActivityViewModelFactory not found

along side with

    /.../build/generated/source/kapt/debug/.../Hilt_ABCApplication.java:20: error: cannot find symbol
    return DaggerABCApplication_HiltComponents_SingletonC.builder()
             ^
    symbol: variable DaggerABCApplication_HiltComponents_SingletonC
    1 error

Using

    hilt_android_version = '2.34.1-beta'
    hilt_view_model_version = "1.0.0-beta01"
    hilt_lifecycle_vm_version = "1.0.0-alpha03"

    classpath "com.google.dagger:hilt-android-gradle-plugin:$hilt_android_version"

    implementation "com.google.dagger:hilt-android:$hilt_android_version"
    kapt "com.google.dagger:hilt-android-compiler:$hilt_android_version"
    implementation "androidx.hilt:hilt-lifecycle-viewmodel:$hilt_lifecycle_vm_version"
    kapt "androidx.hilt:hilt-compiler:$hilt_view_model_version"

**Comment by danysantiago** [comment]
Dupe of https://github.com/google/dagger/issues/2541, specifically see https://github.com/google/dagger/issues/2541#issuecomment-817927472 with regarding to the deprecated hilt-lifecycle-viewmodel artifact.

**Comment by catalin-florescu** [comment]
@danysantiago Well, i've done that yesterday already, but I gave up.

I've done that now again with invalidate cache and restart, but all generated classes are empty, so the app now crash.

    @AggregatedDeps(
        components = "dagger.hilt.android.components.ActivityComponent",
        modules = "package.name.ABCDiModule"
    )
    class package_name_ABCDiModuleModuleDeps {
    }

**Comment by danysantiago** [comment]
Those `@ AggregatedDeps` are meant to be empty. Can you show us the stacktrace of the crash?

Did you remove both `androidx.hilt:hilt-lifecycle-viewmodel` and `androidx.hilt.hilt-compiler` (assuming you are not using `hilt-work`, if you do keep the `androidx.hilt-hilt-compiler` dep).

And just to double check, you did migrate from `@ViewModelInject` to `@HiltViewModel` right?

**Comment by catalin-florescu** [comment]
I use `hilt-work`, i will add back the compiler and try again, also migrated to `@HiltViewModel` when was deprecated.

**Comment by danysantiago** [comment]
Something odd is going on, the `deps` classes are not used at runtime, they are just for propagating information between compilation units. It seems odd you get no compile error but do get this runtime error. Do you know who/what is trying to instantiate `ABCViewModel` ?

Some more things to check / ask:
* Is your `hilt-work` version also `1.0.0-beta01` ?
* Is your Dagger & Hilt versions the same across the rest of your Gradle modules, and in any library module you have you are also applying the compiler?
* Any other error or warning being show that might be the culprit?

**Comment by catalin-florescu** [comment]
Yes, runtime error. I forgot to mention.
Me, i'm trying to instantiate in first activity, splash.

- Is your `hilt-work` version also 1.0.0-beta01 ? **Yes**
- Is your Dagger & Hilt versions the same across... ? **Yes**
- Any other error or warning being show that might be the culprit? **No**, it directs to ViewModel instance directly.

Also I'm trying to instantiate VM using

    ViewModelProvider(this).get(class)

because i have base activity/fragment (if this is relevant).

**Comment by gs-ts** [comment]
I had a similar issue and solved it this way.

initially, in my project's gradle file I had the following:
```
dependencies {
        classpath "com.android.tools.build:gradle:4.1.3"

        classpath "com.google.dagger:hilt-android-gradle-plugin:2.33-beta"
    }
```

and in my app's gradle file I had these hilt dependencies:
```
plugins {
    id 'com.android.application'
    id 'kotlin-android'
    id 'kotlin-kapt'
    id 'dagger.hilt.android.plugin'
}

    // Hilt, dependency injection
    implementation "com.google.dagger:hilt-android:2.33-beta"
    kapt "com.google.dagger:hilt-android-compiler:2.33-beta"
    // Hilt extensions
    implementation "androidx.hilt:hilt-lifecycle-viewmodel:1.0.0-alpha03"
    kapt "androidx.hilt:hilt-compiler:1.0.0-beta01"
```
IF I just update the hilt version from `2.33-beta` to `2.34.1-beta` 
THEN I get the error: `cannot access DefaultActivityViewModelFactory`

IF I update the hilt version from `2.33-beta` to `2.34.1-beta` AND remove  the hilt extensions: `implementation "androidx.hilt:hilt-lifecycle-viewmodel:1.0.0-alpha03"` and `kapt "androidx.hilt:hilt-compiler:1.0.0-beta01"` 
THEN it works.

(invalidate caches and restart was not needed in my case)

Questions:
1. is it safe to remove the hilt extensions to solve this issue?
2. is there a case that these hilt extensions might be needed and if yes then how the above issue will be solved?

**Comment by danysantiago** [comment]
@catalin-florescu, thanks, it seems now you are hitting the 'default' can't provide a `ViewModel`. without a default constructor. This is the usual error message shown on Activities and Fragments using the default view model factory. This can happen if the `ViewModelProvider` is not using Hilt's ViewModelFactory. Make sure you annotate your fragment / activity with `@AndroidEntryPoint` and if this is a base class then, it's children should be Android entry points. However, I'm surprise it stopped working from  `2.33` to `2.34` since the relevant change done between those two versions was to remove the backwards compatibility we had for `androidx.hilt:hilt-lifecycle-viewmodel`.

@gs-ts, yes - It is safe to remove the `hilt-lifecycle-viewmodel` dependency since it was replaced by integrating the `ViewModel` support into Hilt and via `@HiltViewModel`. The artifact `hilt-lifecycle-viewmodel` will no longer be released and is not needed anymore at all. However, the `hilt-compiler` is still needed if you use **other** androidx extensions, such as [`hilt-work`](https://developer.android.com/training/dependency-injection/hilt-jetpack#workmanager).

To recap, for injecting ViewModels in Hilt:
- Annotate your VM class with `@HiltViewModel` and not `@ViewModelInject`
- Remove the `hilt-lifecycle-viewmodel` dependency from all your modules.
- Make sure you have the Hilt compiler in all your modules using Hilt annotations (`com.google.dagger:hilt-compiler`)
- Annotate your activity / fragment with `@AndroidEntryPoint` and use them as the `ViewModelStoreOwner` when retrieving your VM via `ViewModelProvider`
- If you are using the AndroidX Navigation library and want to get a ViewModel scoped to a `NavBackStackEntry` then use the `androidx.hilt:hilt-navigation` or `androidx.hilt:hilt-navigation-fragment` artifacts that have functions for getting the `HiltViewModelFactory` for a given `NavBackStackEntry`.
